sdk/keyvault/keyvault-certificates/src/identifier.ts

```diff
@@ -76,16 +76,17 @@ export class KeyVaultCertificatesIdentifier implements ParsedKeyVaultCertificate
       "certificates",
       "deletedcertificates"
     ];
-    const collection = collections.filter((x) => url.split("/").includes(x))[0];
+    const urlParts = url.split("/");
+    const collection: KeyVaultCertificatesIdentifierCollectionName =
```

Comment:
While I don't think we should be so strict about parsing the collection out, lying (if not matching `/(deleted)?certificates/`) seems wrong. Should you maybe throw? I could live with this change - they are using the wrong one for the given ID - but this was just to be a helper.

Reply:
Yes, I can throw if it doesn't match.